### ISSUE TITLE
Using a CMake module with lots of generated files gives an error "Argument list too long"

### ISSUE THREAD

**Issue by tmakij** [mesonbuild/meson#11691]
**Describe the bug**
Importing a CMake module, that passes many generated files to a custom command, creates a ninja build file that fails with the following error when run:

`Generating subprojects/big_project/custom_tgt_0 with a custom commandninja: fatal: posix_spawn: Argument list too long`

**To Reproduce**
1. Clone the repository from <https://github.com/tmakij/cmake_too_many_generated_files>.
2. `meson setup builddir`
3. `cd builddir`
4. `ninja`

**Expected behavior**
Meson generates the custom command in such a way that ninja does not die. This is how CMake behaves. You can test the CMake version also, it is included in the repository.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)? **Native build**
* what operating system **Arch Linux**
* what Python version are you using **3.10.10**
* what `meson --version` **1.0.1**
* what `ninja --version` if it's a Ninja build **1.11.1**
* cmake --version **3.26.3**

**Comment by eli-schwartz** [comment]
This seemingly only happens when add_custom_command() does not take the output files as arguments, but does claim to create them.

- In the generated cmake project, it simply runs `true` and lists 5000 files as ninja outputs.
- In the generated meson project, it runs `meson --internal cmake_run_ctgt -o <list-of-all-outputs> -O <list-of-all-outputs> -d <workdir> true ';;;'` and then copies each of these files from the hidden cmake build directory into the meson build directory.

Seems like if we're going to be custom handling big lists of source files we need to support RSP files for it.

On the other hand, the use case of "add_custom_command() that generates tons of output files without listing them on the command line" seems... contrived. :)

**Comment by tmakij** [comment]
I created the repository as a minimal reproducible example. I faced the issue originally when I tried to import godot-cpp, which generates a lot of files with a python script, see <https://github.com/godotengine/godot-cpp/blob/feaba551b5a5b2d13ad1c3fdd8c90e67c67ff37c/CMakeLists.txt#L123>. My repository still triggers the argument list too long error, even if the command makes no sense. Hopefully the root cause is the same here.

**Comment by eli-schwartz** [comment]
Yeah, I assume the root cause is the same. Either way, it's definitely a bug of some sort.

**Comment by SanitoGonzalez** [comment]
Any updates? Facing the same issue while trying to import godot-cpp with `cmake.subproject()`.
meson v1.3.1
cmake 3.27.7
